hazelcast/src/main/java/com/hazelcast/spi/impl/proxyservice/impl/ProxyRegistry.java

```diff
@@ -91,16 +90,8 @@ boolean contains(String name) {
      * @param result The ProxyInfo of all proxies in this registry.
      */
     public void getProxyInfos(Collection<ProxyInfo> result) {
-        for (Map.Entry<String, DistributedObjectFuture> entry : proxies.entrySet()) {
-            final DistributedObjectFuture future = entry.getValue();
-            if (!future.isSet()) {
```

Comment:
I guess the check was there because when the future is not done then it's not possible to tell  whether the object is `InitializingObject` or not. 

yes, this very bug will be fixed just by removing the check. I can do that to stay on a safe side + create a new issue to investigate why there was this check at all.